drivers/lcd/lcd_dev.c

```diff
@@ -0,0 +1,274 @@
+      {
+        *((FAR int *)arg) = priv->lcd_ptr->getcontrast(priv->lcd_ptr);
+      }
+      break;
+    case LCDDEVIO_SETCONTRAST:
+      {
+        ret = priv->lcd_ptr->setcontrast(priv->lcd_ptr, (unsigned int)arg);
+      }
+      break;
+    case LCDDEVIO_GETPLANEINFO:
+      {
+        *((FAR struct lcd_planeinfo_s *)arg) = priv->planeinfo;
+      }
+      break;
+    case LCDDEVIO_GETVIDEOINFO:
+      {
+        ret = priv->lcd_ptr->getvideoinfo(priv->lcd_ptr,
+                                          (FAR struct fb_videoinfo_s *)arg);
+      }
+      break;
+    case LCDDEVIO_SETPLANENO:
+      {
+        ret = priv->lcd_ptr->getplaneinfo(priv->lcd_ptr, (int)arg,
+                                          &priv->planeinfo);
+      }
+      break;
+#ifdef CONFIG_FB_CMAP
+    case LCDDEVIO_GETCMAP:
+      {
+        FAR struct fb_cmap_s *cmap = (FAR struct fb_cmap_s *)arg;
+
+        ret = priv->lcd_ptr->getcmap(priv->lcd_ptr, cmap);
+      }
+      break;
+    case LCDDEVIO_PUTCMAP:
+      {
+        FAR const struct fb_cmap_s *cmap = (FAR const struct fb_cmap_s *)arg;
+
+        ret = priv->lcd_ptr->putcmap(priv->lcd_ptr, cmap);
+      }
+      break;
+#endif
+#ifdef CONFIG_FB_HWCURSOR
+    case LCDDEVIO_GETCURSOR:
+      {
+        FAR struct fb_cursorattrib_s *attrib =
+            (FAR struct fb_cursorattrib_s *)arg;
+
+        ret = priv->lcd_ptr->getcursor(priv->lcd_ptr, attrib);
+      }
+      break;
+    case LCDDEVIO_SETCURSOR:
+      {
+        FAR struct fb_setcursor_s *settings =
+            (FAR struct fb_setcursor_s *)arg;
+
+        ret = priv->lcd_ptr->setcursor(priv->lcd_ptr, settings);
+      }
+      break;
+#endif
+    default:
+      ret = -EINVAL;
+      break;
+    }
+
+  return ret;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: lcddev_register
+ *
+ * Description:
+ *   Register the LCD character driver as /dev/lcdN.
+ *
+ * Input Parameters:
+ *   devno - The LCD device number.
+ *
+ * Returned Value:
+ *   Zero (OK) is returned on success.  Otherwise a negated errno value is
+ *   returned to indicate the nature of the failure.
+ *
+ ****************************************************************************/
+
+int lcddev_register(int devno)
+{
+  FAR struct lcddev_dev_s *priv;
+  int ret = OK;
+  char devname[16];
+
+  /* Allocate a new lcd_dev driver instance */
+
+  priv = (FAR struct lcddev_dev_s *)kmm_zalloc(sizeof(struct lcddev_dev_s));
+
+  if (!priv)
+    {
+      return -ENOMEM;
+    }
+
+  priv->lcd_ptr = board_lcd_getdev(devno);
+  ret = priv->lcd_ptr->getplaneinfo(priv->lcd_ptr, 0, &priv->planeinfo);
+  if (ret < 0)
+    {
+      goto err;
+    }
+
+  snprintf(devname, sizeof(devname), "/dev/lcd%i", devno);
+  ret = register_driver(devname, &lcddev_fops, 0666, priv);
+
```

Comment:
remove the blank line?

Reply:
between register_driver and if? why? this does not violate the coding standard

Reply:
just suggestion, because normally, there isn't blank between ret = and if (ret < 0)

Reply:
ok, if prefer it I'll change it.